### PR TITLE
return a NamedTuple, not a Matrix from predict

### PR DESCRIPTION
**Pull request by mkborregaard**
As discussed on Slack.
cc @kleinschmidt @nalimilan @andreasnoack

See also https://github.com/JuliaStats/StatsModels.jl/pull/81.

### CODE REVIEW

File: src/lm.jl [hunk 1]
```diff
@@ -220,7 +220,7 @@ function predict(mm::LinearModel, newx::AbstractMatrix;
         error("only :confidence and :prediction intervals are defined")
     end
     retinterval = quantile(TDist(dof_residual(mm)), (1. - level)/2) * sqrt.(retvariance)
-    hcat(retmean, retmean .+ retinterval, retmean .- retinterval)
+    (newy = retmean, lower = retmean .+ retinterval, upper = retmean .- retinterval)
```
Comment (nalimilan) [inline src/lm.jl]: Just `y` would look better IMHO.

Comment (mkborregaard) [reply src/lm.jl]: fixed

Comment (nalimilan) [reply src/lm.jl]: FWIW, R uses `fit`, `lwr` and `upr`. The two latter are clearly bad, but `fit` could also work. Opinions?

Comment (kleinschmidt) [reply src/lm.jl]: I don't like "fit" actually, since this is a prediction from new data, not for fitted values.

Comment (mkborregaard) [reply src/lm.jl]: mean?

Comment (mkborregaard) [reply src/lm.jl]: value is not bad either

**Comment by mkborregaard** [comment]
I've updated the tests too now. Travis test error is unrelated.

File: src/lm.jl (hunk 1 above)
Comment (mschauer) [reply src/lm.jl]: Sorry, what kind of interval does this return when the caller provides `:prediction`?

Comment (mkborregaard) [reply src/lm.jl]: The prediction interval

Comment (mschauer) [reply src/lm.jl]: Can you elaborate? Does this have to do with taking or not taking uncertainty in the estimated parameters into account?

Comment (mkborregaard) [reply src/lm.jl]: It's the 95% confidence interval on unobserved data points, whereas the :confidence interval is the confidence on the fit. So, it incorporates the variance in the data itself

Comment (mkborregaard) [reply src/lm.jl]: In Bayesian stats you'd get it by simulating new values from the posterior and taking the (0.025, 0.975)  percentiles

Comment (mschauer) [reply src/lm.jl]: Thanks, rephrasing in my own words `:confidence:` gives the confidence interval for the location parameter of the conditional distribution of the response and `:prediction` gives an interval which contains the unobserved response variable with a given probability

Comment (mkborregaard) [reply src/lm.jl]: Yes exactly except that those words are Bayesian

Comment (nalimilan) [reply src/lm.jl]: Let's go with `value` then?

Comment (andreasnoack) [reply src/lm.jl]: I think I prefer `prediction`.

Comment (kleinschmidt) [reply src/lm.jl]: I also prefer prediction even if it's a little verbose.  in my own code I tend to go for `pred` but I understand that that is a convention that works for me but maybe not everyone.

Comment (nalimilan) [reply src/lm.jl]: Fine with me.

Comment (mkborregaard) [reply src/lm.jl]: Great! The only possibly problem with that (and argument for pred or value) is that we now have `:confidence` and `:prediction` intervals, both of which return a variable named `prediction`

Comment (mkborregaard) [reply src/lm.jl]: sorry to be so wavering :-/

Comment (nalimilan) [reply src/lm.jl]: Fine with me.

Comment (mkborregaard) [reply src/lm.jl]: OK it's `prediction` now and tests should pass so feel free to merge when you're happy.

File: test/runtests.jl [hunk 2]
```diff
@@ -498,7 +498,7 @@ end
     @test sprint(show, ftest(mod, nullmod)) ==
         """
                 Res. DOF DOF ΔDOF    SSR    ΔSSR      R²    ΔR²       F* p(>F)
-        Model 1       10   3      0.1283          0.9603
+        Model 1       10   3      0.1283          0.9603
```
Comment (mkborregaard) [inline test/runtests.jl]: oh wow, Atom is really aggressive on whitespace

**Comment by nalimilan** [comment]
So this should break with the current StatsModels, meaning we should first tag a new release, and then require it from this PR, right?

**Comment by mkborregaard** [comment]
Yes the StatsModels PR is harmless but this one requires it

**Pull request merged**